config/software/go.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Copyright 2019 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name "go"
+default_version "1.13.1"
+license "BSD-3-Clause"
+license_file "https://raw.githubusercontent.com/golang/go/master/LICENSE"
+
+# Check golang releases at: https://golang.org/dl/
+url_template = "https://dl.google.com/go/go#{version}.%{platform}-%{arch}.tar.gz"
+
+# Default architecture
+arch = "amd64"
+
+if windows?
+  platform = "windows"
+
+  # On windows Go uses either '.zip' or '.msi'
+  url_template = "https://dl.google.com/go/go#{version}.%{platform}-%{arch}.zip"
+
+  version "1.13.1" do
+    source sha256: "24cb08d369c1962cccacedc56fd79dc130f623b3b667a316554621ad6ac9b442",
+           url: url_template % { platform: platform, arch: arch }
+  end
+elsif mac_os_x?
+  platform = "darwin"
+
+  version "1.13.1" do
+    source sha256: "f3985fced3adecb62dd1e636cfa5eb9fea8f3e98101d9fcc4964d8f1ec255b7f",
+           url: url_template % { platform: platform, arch: arch }
+  end
+else
+  platform = "linux"
+
+  version "1.13.1" do
+    source sha256: "94f874037b82ea5353f4061e543681a0e79657f787437974214629af8407d124",
+           url: url_template % { platform: platform, arch: arch }
+  end
+end
+
+build do
+  mkdir "#{install_dir}/embedded/bin"
+  %w{go gofmt}.each do |bin|
+    link "#{project_dir}/go/bin/#{bin}", "#{install_dir}/embedded/bin/#{bin}"
```

Comment:
I think you'll want to do a `copy` or `sync` here - otherwise you'll be linking from the "build" directory to the installation directory, resulting in failed linking when it's packaged up.

Reply:
Thanks @scotthain for the review. As you can see in the commit history, I originally did a sync but it was causing failures (reference: https://github.com/chef/omnibus-software/pull/1098/commits/3fc04fcff026ee98b1d457eab993a187990337f9). Another thing to consider is that we need Go to compile the binaries we will be developing but we don’t need to package it. So probably we will delete these links.

Reply:
Ah in that case I don't think it belongs in `omnibus-software` since your use case is slightly different. Since you'll be deleting the links in (I assume) a separate, private (by that I mean not-in-omnibus-software) "cleanup" definition, this will break if anyone assumes it "installs go". @chef/omnibus-maintainers  any thoughts on this?

Reply:
I was going to add that cleanup in this repo, I am not quite done with this PR, I’m still doing compile tests. I added it here because Rust is also here and I thought it would be a good place to add Go, but if this is not the correct repo, where should you recommend I should add it?

Reply:
So the rust def and corresponding uninstall is slightly different because it actually does a full install of rust to the `install_dir` and then does a full uninstall. The way I see it there are 2 options:
- Do a full install of go in this def and create a `go-uninstall` def that does  a full remove. This allows folks who actually want to include/have a "permanent" go installation work correctly when using this def. (and a corresponding uninstall)
- Put the def as is (with the linking) in the project that uses it, and remove the links in a cleanup def in the project that uses it.

Happy to use mouthwords to talk more about this if I'm not being clear.

Reply:
Sounds good to me. Thanks for the review Scott! ❤️ 🎉 🥇

Reply:
Absolutely! Thanks for being open to the feedback 🚀 🍰